### PR TITLE
Test pallet order

### PR DESCRIPTION
**Pull request by claravanstaden**
Adds a test to verify that the Ethereum Outbound Queue digest is in contained in a block.

I used @yrong's PR https://github.com/Snowfork/snowbridge/pull/1126/files to use the pallet order to finalize the blocks. I verified if I change the runtime pallet order so that the MessageQueue pallet is before the EthereumOutboundQueue pallet, the test fails.

### CODE REVIEW

File: bridges/snowbridge/runtime/test-common/src/lib.rs [hunk 1]
```diff
@@ -193,9 +195,104 @@ pub fn send_transfer_token_message_success<Runtime, XcmConfig>(
...
 			let digest_items = digest.logs();
 			assert!(digest_items.len() == 1 && digest_items[0].as_other().is_some());
+
+			let digest_item = &digest_items[0];
+			match digest_item {
+				DigestItem::PreRuntime(_, _) => panic!("is PreRuntime"),
+				DigestItem::Consensus(_, _) => panic!("is Consensus"),
+				DigestItem::Seal(_, _) => panic!("is Seal"),
+				DigestItem::Other(veccc) => {
+					assert_eq!(veccc[0], 0);
```
Comment (alistair-singh) [inline bridges/snowbridge/runtime/test-common/src/lib.rs:207]: I do not understand this assert, why assert only the first element?

File: bridges/snowbridge/runtime/test-common/src/lib.rs [hunk 2]
```diff
@@ -193,9 +195,104 @@ pub fn send_transfer_token_message_success<Runtime, XcmConfig>(
...
+				RuntimeHelper::<Runtime, AllPalletsWithoutSystem>::run_to_block_with_finalize(
+					next_block_number.as_u32(),
+				);
+			let digest = included_head.digest();
+			let digest_items = digest.logs();
+
+			let mut found_outbound_digest = false;
+			for digest_item in digest_items {
+				match digest_item {
+					DigestItem::Other(_) => found_outbound_digest = true,
```
Comment (alistair-singh) [inline bridges/snowbridge/runtime/test-common/src/lib.rs:290]: So we assert that the message is queued on a block and then the DigestItem exists in the next block. This test asserts that the pallet order is correct?

**Review by alistair-singh — APPROVED** [review]
Just some comments to help my understanding of the test. But looks good.

**Review by vgeddes — APPROVED** [review]

**Review by yrong — APPROVED** [review]
+1

File: bridges/snowbridge/runtime/test-common/src/lib.rs (hunk 1 above)
Comment (claravanstaden) [reply bridges/snowbridge/runtime/test-common/src/lib.rs]: Whoops, good catch. I tested something locally and forgot to revert this change. Reverted here: https://github.com/Snowfork/polkadot-sdk/pull/112/commits/b570e54a412f27ff4e157408ac365d17e5c91e90

File: bridges/snowbridge/runtime/test-common/src/lib.rs (hunk 2 above)
Comment (claravanstaden) [reply bridges/snowbridge/runtime/test-common/src/lib.rs:279]: Yes, so if the pallet order is incorrect (i.e. the `MessageQueue` pallet is before the `EthereumOutboundQueue` pallet in the `construct_runtime` macro), our message digest won't be found. I tested this locally, works nicely.

**Pull request merged**